pkcs11/_pkcs11.pyx

```diff
@@ -247,7 +254,12 @@ class Token(types.Token):
 
         assertRV(_funclist.C_OpenSession(self.slot.slot_id, flags, NULL, NULL, &handle))
 
-        if pin is not None:
+        if p_auth:
+            if self.flags & TokenFlag.PROTECTED_AUTHENTICATION_PATH:
+                assertRV(_funclist.C_Login(handle, user_type, NULL, < CK_ULONG > 0))
+            else:
+                raise ArgumentsBad('Protected authentication is not supported by loaded module')
```

Comment:
Please use double quotes for human facing strings (like error messages).

Reply:
Will do